### PR TITLE
udpate to work on stable

### PR DESCRIPTION
**Pull request by Gankro**
r? @bluss

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -777,10 +822,20 @@ impl<T> IntoIterator for LinkedList<T> {
...
+
+
+
+
+
+
+
+
+
+
```
Comment (bluss) [inline src/lib.rs]: These 8 lines, is that a new style you're using? :)

File: src/lib.rs [hunk 2]
```diff
@@ -705,29 +718,61 @@ impl<A> Extend<A> for LinkedList<A> {
...
+            match (a.next(), b.next()) {
+                (Some(x), Some(y)) => match x.cmp(&y) {
+                    Ordering::Equal => {}
+                    otherwise => return otherwise,
+                },
+                (None, None) => return Ordering::Equal,
+                (None, _) => return Ordering::Less,
+                (_, None) => return Ordering::Greater,
+            }
+        }
```
Comment (bluss) [inline src/lib.rs]: This one could be partial_cmp().unwrap() right in theory?

File: src/lib.rs [hunk 3]
```diff
@@ -398,7 +409,8 @@ impl<'a, T> Cursor<'a, T> {
                 Some(next) => {
                     self.prev = Raw::some(&mut **next);
                     unsafe {
-                        Some(mem::copy_mut_lifetime(self, &mut next.elem))
+                        // upgrade the lifetime
+                        Some(mem::transmute(&mut next.elem))
```
Comment (bluss) [inline src/lib.rs]: You can upgrade lifetimes without transmute, it's a bit wordy though! `&mut *(&mutnext.elem as *mut _)`   something like that. Either way is fine to me, since it's a bit much.

**Comment by bluss** [comment]
Lol, flash cat has a resemblance to you!

**Comment by bluss** [comment]
Great, r=me when you want. Act on comments if you want to.

File: src/lib.rs (hunk 1 above)
Comment (Gankro) [inline src/lib.rs]: Yeah I like to make big spaces for "this is something different".

File: src/lib.rs (hunk 2 above)
Comment (Gankro) [inline src/lib.rs]: ...yes

File: src/lib.rs (hunk 3 above)
Comment (Gankro) [inline src/lib.rs]: Yeah this is something that needs work... not today.

**Comment by Gankro** [comment]
I have many faces

**Comment by bluss** [comment]
hehe wow, what is this, a map of all you pseudonyms?

**Comment by Gankro** [comment]
It's various avatars that were made for me, with the names of the authors (and their lineage). From back in my MSPA days.
